flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnConfigurationITCase.java

```diff
@@ -187,6 +191,12 @@ public void testFlinkContainerMemory() throws Exception {
 					assertThat(
 						(double) taskManagerInfo.getHardwareDescription().getSizeOfJvmHeap() / (double) expectedHeadSize,
 						is(closeTo(1.0, 0.15)));
+
+					final int defaultTaskManagerMemoryMB = ConfigurationUtils.getTaskManagerHeapMemory(configuration).getMebiBytes();
```

Comment:
I kind of think it differently. 
I think we already have `testCreateSlotsPerWorker` that checks the behavior of `updateTaskManagerConfigAndCreateWorkerSlotProfiles`. If we just set constant `defaultTaskManagerMemoryMB` and expect a corresponding constant `expectedManagedMemoryMB`, then later if we change the logic in `updateTaskManagerConfigAndCreateWorkerSlotProfiles `  that it calculates managed memory from task manager memory differently, we may have to also change this test case.

Reply:
True, both tests have to be changed, more effort. On the other side, this brings attention to them and forces to check the change against any previous assumptions expressed by tests. In this case, the real computation might change in a way that `updateTaskManagerConfigAndCreateWorkerSlotProfiles` is not used at all etc.

Reply:
I tend to disagree. I don't think it is necessary for this test case to have any assumption on how `updateTaskManagerConfigAndCreateWorkerSlotProfiles` calculates managed memory. The purpose of this test case, to my understanding, is to validate that configurations are well passed to TMs rather than the calculation logics. The latter is already covered by another test case. If it does happen that the real calculation changes in a way that `updateTaskManagerConfigAndCreateWorkerSlotProfiles` is not used any more, then this test case should also fail as long as the calculation on TM side changes consistently, or otherwise the other test case `testCreateSlotsPerWorker` should fail.

Reply:
I tend to agree with @xintongsong here. Tests ideally test one thing targeted, and the other aspect is already covered.

Reply:
I think it's acceptable to depend on `updateTaskManagerConfigAndCreateWorkerSlotProfiles` here.

Reply:
ok, agreed
it is used more like a helper function to test another thing then